### PR TITLE
added ecs checks to foundations policy

### PR DESCRIPTION
**Pull request by amanenk**

### CODE REVIEW

File: queries/ecs/task-definitions-secure-networking.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+SELECT t.account_id, t.region, t.arn, cd.name
+FROM aws_ecs_task_definitions t
+         JOIN aws_ecs_task_definition_container_definitions cd ON t.cq_id = cd.task_definition_cq_id
+WHERE t.network_mode = 'host'
+  AND (cd.privileged IS DISTINCT FROM TRUE OR cd."user" = 'root');
```
Comment (disq) [inline queries/ecs/task-definitions-secure-networking.sql:5]: > This control checks whether an active Amazon ECS task definition that has host networking mode also has privileged or user container definitions. The control fails for task definitions that have host network mode and container definitions where privileged=false or is empty and user=root or is empty.

"user=root" is checked, but not the "is empty"

**Review by disq — CHANGES_REQUESTED** [review]

File: queries/ecs/task-definitions-secure-networking.sql (hunk 1 above)
Comment (bbernays) [reply queries/ecs/task-definitions-secure-networking.sql:5]: Won't `IS DISTINCT FROM TRUE` find all values that are not True (so null or false)?

**Review by bbernays — APPROVED** [review]
LGTM

File: queries/ecs/task-definitions-secure-networking.sql (hunk 1 above)
Comment (disq) [reply queries/ecs/task-definitions-secure-networking.sql:5]: > The control fails for task definitions that have host network mode and container definitions where privileged=false or is empty and user=root or is empty.

The control fails for :
 -> task definitions have host network mode
AND
 -> container definitions where privileged=false or is empty (this one is covered by `IS DISTINCT FROM TRUE`)
AND
 -> user=root or is empty (so I think a `cd."user" is null` needs to be added?)

at least that's how I parse that complex definition...

**Review by disq — CHANGES_REQUESTED** [review]
Another nit sorry didn't notice earlier: Filenames in queries/ should be separated with underscore and not `-`, as that's the way it's been done previously...

`ecs-services-with-public-ips.sql` => `ecs_services_with_public_ips.sql`
`task-definitions-secure-networking.sql` => `task_definitions_secure_networking.sql`

**Review by disq — APPROVED** [review]

**Review by spangenberg — APPROVED** [review]
LGTM

**Pull request merged**